pkg/daemon/manager/manager.go

```diff
@@ -150,6 +151,37 @@ func (m *manager) EnsureServiceState(service winsvc.Service, state svc.State) er
 	return winsvc.WaitForState(service, state)
 }
 
+// Stop the service, and wait for the process associated with the service to stop
+func (m *manager) stopServiceAndProcess(winSvc *mgr.Service) error {
+	status, err := winSvc.Query()
+	if err != nil {
+		return errors.Wrap(err, "error querying service")
+	}
```

Comment:
Can't you just return here `if status.ProcessId == 0`. The code will read much easier then.

Reply:
Unfortunately no, the service itself needs to be stopped.
The stopping of the service needs to be sandwiched between opening the process and waiting/stopping the process

Reply:
So there could be a running service without a process associated with. That does not sound right. What happens if you kill the process? The service stays at running?

Reply:
> So there could be a running service without a process associated with

I don't know if its possible but that edge case needs to be handled.

> What happens if you kill the process

Killing the process without sending a stop signal for the service will be handled dependent on the service configuration, ideally the service will restart the process.

Reply:
>ideally the service will restart the process.

I know that does not happen with our current service configuration. The services stay stopped when the process crashes.